pkg/filter/headertojson/headertojson.go

```diff
@@ -0,0 +1,136 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package headertojson
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+
+	json "github.com/goccy/go-json"
+	"github.com/megaease/easegress/pkg/context"
+	"github.com/megaease/easegress/pkg/object/httppipeline"
+)
+
+const (
+	// Kind is the kind of Kafka
+	Kind = "HeaderToJSON"
+
+	resultJSONEncodeDecodeErr = "JSONEncodeDecodeErr"
+)
+
+func init() {
+	httppipeline.Register(&HeaderToJSON{})
+}
+
+type (
+	// HeaderToJSON is make http request header to json
+	HeaderToJSON struct {
+		filterSpec *httppipeline.FilterSpec
+		spec       *Spec
+		headerMap  map[string]string
+	}
+)
+
+var _ httppipeline.Filter = (*HeaderToJSON)(nil)
+
+// Kind return kind of HeaderToJSON
+func (h *HeaderToJSON) Kind() string {
+	return Kind
+}
+
+// DefaultSpec return default spec of HeaderToJSON
+func (h *HeaderToJSON) DefaultSpec() interface{} {
+	return &Spec{}
+}
+
+// Description return description of HeaderToJSON
+func (h *HeaderToJSON) Description() string {
+	return "HeaderToJSON convert http request header to json"
+}
+
+// Results return possible results of HeaderToJSON
+func (h *HeaderToJSON) Results() []string {
+	return []string{resultJSONEncodeDecodeErr}
+}
+
+func (h *HeaderToJSON) init() {
+	h.headerMap = make(map[string]string)
+	for _, header := range h.spec.HeaderMap {
+		h.headerMap[header.Header] = header.JSON
+	}
+}
+
+// Init init HeaderToJSON
+func (h *HeaderToJSON) Init(filterSpec *httppipeline.FilterSpec) {
+	h.filterSpec, h.spec = filterSpec, filterSpec.FilterSpec().(*Spec)
+	h.init()
+}
+
+// Inherit init HeaderToJSON based on previous generation
+func (h *HeaderToJSON) Inherit(filterSpec *httppipeline.FilterSpec, previousGeneration httppipeline.Filter) {
+	previousGeneration.Close()
+	h.Init(filterSpec)
+}
+
+// Close close HeaderToJSON
+func (h *HeaderToJSON) Close() {
+}
+
+// Status return status of HeaderToJSON
+func (h *HeaderToJSON) Status() interface{} {
+	return nil
+}
+
+func (h *HeaderToJSON) encodeJSON(input map[string]interface{}) ([]byte, error) {
+	return json.Marshal(input)
+}
+
+func (h *HeaderToJSON) decodeJSON(ctx context.HTTPContext) (map[string]interface{}, error) {
+	ans := make(map[string]interface{})
+
+	b, err := io.ReadAll(ctx.Request().Body())
+	if err != nil {
+		return nil, err
+	}
+	json.Unmarshal(b, &ans)
+	return ans, nil
+}
+
+// Handle handle HTTPContext
+func (h *HeaderToJSON) Handle(ctx context.HTTPContext) string {
+	result := h.handle(ctx)
+	return ctx.CallNextHandler(result)
+}
+
+func (h *HeaderToJSON) handle(ctx context.HTTPContext) string {
+	ans, err := h.decodeJSON(ctx)
+	if err != nil {
+		return resultJSONEncodeDecodeErr
+	}
+	for header, json := range h.headerMap {
+		value := ctx.Request().Header().Get(http.CanonicalHeaderKey(header))
+		ans[json] = value
+	}
```

Comment:
1. propose to do `http.CanonicalHeaderKey` in `Init`.
2. if the request does not contain any required header, could we avoid the JSON decoding and encoding?